src/nni_manager/core/nnimanager.ts

```diff
@@ -344,24 +338,138 @@ class NNIManager implements Manager {
         const execDuration: number = this.experimentProfile.execDuration;
         for (; ;) {
             await delay(1000 * 60 * 10); // 10 minutes
-            this.experimentProfile.execDuration = execDuration + (Date.now() - startTime) / 1000;
+            this.experimentProfile.execDuration = execDuration + (Date.now() - startTime) / 1000 - this.suspendDuration;
             await this.storeExperimentProfile();
         }
     }
 
+    private async requestTrialJobsStatus(): Promise<number> {
+        const deferred: Deferred<number> = new Deferred<number>();
+        let finishedTrialJobNum: number = 0;
+        for (const trialJobId of Array.from(this.trialJobs.keys())) {
+            const trialJobDetail: TrialJobDetail = await this.trainingService.getTrialJob(trialJobId);
+            const oldTrialJobDetail: TrialJobDetail | undefined = this.trialJobs.get(trialJobId);
+            assert(oldTrialJobDetail);
+            if (oldTrialJobDetail !== undefined && oldTrialJobDetail.status !== trialJobDetail.status) {
```

Comment:
assert(oldTrialJobDetail);  and oldTrialJobDetail !== undefined is duplicated. Need to leave just one.

Reply:
agree, but if I remove `oldTrialJobDetail !== undefined`, oldTrialJobDetail.status might be undefined, does not pass compiling.

Reply:
removed assert.